### PR TITLE
Switch abstraction layer from SYCL buffers to device USM pointers

### PR DESCRIPTION
**Pull request by michael-smirnov**
This is a continuation of #1211

### CODE REVIEW

File: examples/oneapi/dpc/makefile_lnx [hunk 1]
```diff
@@ -69,19 +69,25 @@ ifndef TBBROOT
...
 else
     override threading = parallel
     THREADING_LIB := onedal_thread
 endif
 
+ifeq ($(RES_EXT),so)
+    ONEDAL_LIBS := -Wl,-Bdynamic -lonedal_dpc \
+                   -Wl,-Bdynamic -lonedal_core \
+                   -Wl,-Bdynamic -l$(THREADING_LIB) \
+                   -Wl,"$(DAAL_PATH)"/libonedal_sycl.a
```
Comment (rlnx) [inline examples/oneapi/dpc/makefile_lnx:83]: `-Wl,-Bdynamic` and just `-Wl` can be removed

File: examples/oneapi/dpc/makefile_lnx [hunk 2]
```diff
@@ -69,19 +69,25 @@ ifndef TBBROOT
...
+else
+    ONEDAL_LIBS := -Wl,"$(DAAL_PATH)"/libonedal_dpc.a \
+                   -Wl,"$(DAAL_PATH)"/libonedal_core.a \
+                   -Wl,"$(DAAL_PATH)"/lib$(THREADING_LIB).a \
+                   -Wl,"$(DAAL_PATH)"/libonedal_sycl.a
```
Comment (rlnx) [inline examples/oneapi/dpc/makefile_lnx:88]: `-Wl` can be removed

File: examples/oneapi/dpc/makefile_lnx [hunk 3]
```diff
@@ -91,10 +97,7 @@ COPTS := -std=c++17 \
...
          -I"$(DAALROOT)/include"
 
-LIBS := -Wl,$(-B) -lonedal_dpc \
-        -Wl,$(-B) -lonedal_core \
-        -Wl,$(-B) -lonedal_sycl \
-        -Wl,$(-B) -l$(THREADING_LIB) \
+LIBS := $(ONEDAL_LIBS) \
         -Wl,-Bdynamic -ltbb \
         -Wl,-Bdynamic -ltbbmalloc \
         -Wl,-Bdynamic -lpthread \
```
Comment (rlnx) [inline examples/oneapi/dpc/makefile_lnx:103]: `-Wl,-Bdynamic` can be removed

File: cpp/oneapi/dal/test/engine/catch_main.cpp [hunk 4]
```diff
@@ -19,6 +19,13 @@
 #include <sstream>
 #include <unordered_map>
 
+// CATCH_CONFIG_POSIX_SIGNAL enables handling of POSIX signals.
+// For unknown reason user-defined handlers for signals
+// (see https://en.wikipedia.org/wiki/C_signal_handling)
+// catches SIGSEGV signal when USM pointer is accessed on host.
+// To make USM work, we disable signal handling in Catch2.
+#define CATCH_CONFIG_NO_POSIX_SIGNALS
+
```
Comment (rlnx) [inline cpp/oneapi/dal/test/engine/catch_main.cpp:28]: Can be deleted. this is already included to `oneapi/dal/test/engine/catch.hpp`

**Review by napetrov — APPROVED** [review]

**Pull request merged**